### ISSUE TITLE
[Bug]: Refresh published and upload new notes -> "Repo empty"-Error

### ISSUE THREAD

**Issue by ArneNostitz** [ObsidianPublisher/obsidian-github-publisher#87]
### Issue validation

- [X] My issue is not a question relative to [Material Mkdocs](https://squidfunk.github.io/mkdocs-material/) nor [Mkdocs](https://www.mkdocs.org/) (*[You can open a discussion to get more help!](https://github.com/squidfunk/mkdocs-material/discussions)*)
- [X] I checked the issue to prevent duplicate
- [X] I checked my configurations files and the documentation

### Workflow

Obsidian plugin 💠

### Describe the bug

Uploading a single file works, uploading with another command "Refresh published and upload new notes" leads to an error
<img width="933" alt="Screenshot 2023-01-23 at 17 12 43" src="https://user-images.githubusercontent.com/16592596/214091918-2279a435-d303-49ad-b28b-a53914e61fe3.png">

after a force reload and trying again it uploaded one new file and deleted!!!! three that were already uploaded (they still have share:true) 

this is not working reliable - and deleting files is fatal!!

### Configuration

-

**Comment by Lisandra-dev** [comment]
1/ I need your configuration and plugin version. This bug must be fixed in the late version (4.11.16)
2/ You can disable auto-clean
3/ You can revert the PR to restore your file.

> this is not working reliable - and deleting files is fatal!!
This is pretty disrespectful to say on a plug-in coded by an alone people for free.

**Comment by ArneNostitz** [comment]
Plugin Version was 4.11.15rc4; but auto-clean was off

ill check again with the current version

**Comment by Lisandra-dev** [comment]
Again, I need a data.json configuration, the repo where you tried to upload and a example note.

**Comment by ArneNostitz** [comment]
with the newest version it does not find changes and new files when hitting "Refresh published and upload new notes"

**Comment by ArneNostitz** [comment]
{
  "githubRepo": "***",
  "githubName": "***",
  "GhToken": "***",
  "githubBranch": "main",
  "shareKey": "share",
  "excludedFolder": [],
  "fileMenu": true,
  "editorMenu": true,
  "downloadedFolder": "fixed",
  "folderDefaultName": "Obsidian/Arne",
  "yamlFolderKey": "",
  "rootFolder": "",
  "workflowName": "",
  "embedImage": true,
  "defaultImageFolder": "",
  "autoCleanUp": false,
  "autoCleanUpExcluded": [],
  "folderNote": false,
  "folderNoteRename": "index.md",
  "convertWikiLinks": false,
  "convertForGithub": false,
  "subFolder": "",
  "embedNotes": false,
  "copyLink": false,
  "mainLink": "",
  "linkRemover": "",
  "hardBreak": false,
  "logNotice": false,
  "convertDataview": false,
  "useFrontmatterTitle": false,
  "censorText": [],
  "inlineTags": false,
  "dataviewFields": [],
  "excludeDataviewValue": [],
  "metadataFileFields": [],
  "frontmatterTitleKey": "title",
  "shareExternalModified": false,
  "automaticallyMergePR": true,
  "metadataExtractorPath": "",
  "convertInternalNonShared": false,
  "frontmatterTitleRegex": "",
  "frontmatterTitleReplacement": "",
  "ExcludedFolder": []
}

**Comment by Lisandra-dev** [comment]
> with the newest version it does not find changes and new files when hitting "Refresh published and upload new notes"

Try to force re-upload, then.

**Comment by ArneNostitz** [comment]
(and sorry if my comment earlier was sounding rude - its just very scary if a plugin deletes files from the server. wasnt meant to sound rude in any way ... its amazing what you do and i really have the most respect for anything that all you amazing plugin people do!!! this is really the coolest community and i am super thankful, that you are so responsive and made this plugin in the first place!!)

**Comment by ArneNostitz** [comment]
uploading single files does work. just the other command is not ... says "no new files"

**Comment by Lisandra-dev** [comment]
It's pretty strange that it deleted file if you use the fixed repository.... I need to check in the code, then

**Comment by Lisandra-dev** [comment]
How works the upload all commands?

**Comment by ArneNostitz** [comment]
i added a few files without the "share" tag manually (as i have a few hundred files and open one by one adding yaml is timeconsuming for those old files) ... upload all tried to delete them all. I quit obsidian

**Comment by Lisandra-dev** [comment]
When trying to debug this, remove the auto-merging pull-request. It will help to prevent restoring in loop the deleted file, while allowing checking the push.

**Comment by Lisandra-dev** [comment]
BTW the file won't be deleted in your vault! Only in your repository. If they don't have the `share:key` and not in your repo, nothing will happend.
Do you have the log message about deleting file ?

**Comment by Lisandra-dev** [comment]
Okay, I think i found the error, could you try to download the beta ? You can do it using BRAT or manually!

**Comment by ArneNostitz** [comment]
I had to leave the office ... will check tomorrow! thanks!

**Comment by Lisandra-dev** [comment]
Hello! Any update ?

**Comment by ArneNostitz** [comment]
I am so sorry. I am running from one meeting to the next … hope I have an hour at the office tomorrow

**Comment by Lisandra-dev** [comment]
Take your time! Good luck at your job :)

**Comment by Lisandra-dev** [comment]
Hello!
I'm sorry, I wait for your message/reply before pushing the beta to prod!

**Comment by ArneNostitz** [comment]
hi, so sorry ... i am on a shoot with 16h days ... i will be back on my obsidian hopefully soon - but could be that i am shooting until march ... 

can you post a link to the beta here so i can download? dont know what "brat" is. 😬 will try to do it on the weekends somehow. cant guarantee thou - really sorry bout that

**Comment by ArneNostitz** [comment]
if its only me having these issues its probably not that pressing

**Comment by QuickFastly** [comment]
This may not be the same bug since I didn't get an error saying the repo was empty, but when I tried to update my repo today, every single note got deleted with no warning and no apparent errors. Disabling auto clean didn't help, uploading single files didn't work. I was initially using version 5.3.1, then tried again after updating to 5.3.2 with the same problem, no new notes were uploaded. Every pull request only contained my attachments, never any .md files, even though the plugin appeared to be working normally, even with notice every error enabled.

I tried downgrading to a few different older versions, same problem every time, until ultimately I landed on 4.12.0 where it finally started working again. I don't have any more information, unfortunately.

Sorry for being unhelpful, but at this point I just want my workflow to _work_ so I'm going to stick with the old version, but wanted to at least confirm that there is in fact something going wrong somewhere.
